autorest/azure/token.go

```diff
@@ -104,6 +154,79 @@ func (spt *ServicePrincipalToken) EnsureFresh() error {
 
 // Refresh obtains a fresh token for the Service Principal.
 func (spt *ServicePrincipalToken) Refresh() error {
+	newToken, err := spt.GetOauthToken()
```

Comment:
It's not clear to me why you moved the logic from Refresh into a separate method. I am guessing you needed the method in some other context? Still, the original design was for Refresh to handle all cases. I prefer that we move the logic back into Refresh.